### DISCUSSION TITLE
Powershell unable to find oh-my-posh module

### DISCUSSION THREAD

**Discussion by awa5114** [JanDeDobbeleer/oh-my-posh#2219] [Troubleshoot] [answered]
I installed `oh-my-posh` using `winget` as described by the [docs](https://ohmyposh.dev/docs/installation/windows#install)..

Then, I tried the following in an administrator powershell terminal:

`import-module oh-my-posh`.

I got an error:

> import-module : The specified module 'oh-my-posh' was not loaded because no valid module file was
found in any module directory.

What else do I need to do besides install with `winget` to be able to import oh-my-posh as a powershell module?

**Comment by philippetev** [comment, marked as answer]
Do you really need it? That module was deprecated recently since almost all of its functions are currently being injected in PowerShell by oh-my-posh itself.

**Reply by awa5114** [reply]
I have my own custom posh prompt which I inject into my powershell session by adding the following to `$PROFILE`:

```
import-module oh-my-posh
set-poshprompt "~\desktop\posh_themes\custom_theme.json"
```

If there's another way to permanently modify my prompt using `$PROFILE` then I'd be open to doing that instead.

**Reply by philippetev** [reply]
How about this:
`oh-my-posh init pwsh -c "$env:USERPROFILE\path\to\custom_theme.json" | Invoke-Expression`
In PowerShell's profile, of course.
Since you have OMP installed via winget, it's already in $PATH.
Replace what you currently have in $PROFILE with this line and you're good to go.

**Reply by JanDeDobbeleer** [reply]
This is the correct answer.

**Reply by awa5114** [reply]
I added that command to my `$PROFILE` and it seems to work. Thank you.

**Comment by ShaguarWKL** [comment]
Hi. Seems like the same issue I encountered basically calling Terminal via different methods can cause OMP path to not be in $PATH for some reason. You don't need to do import-module and set-prompt anymore, winget should be setting all those on install.

https://github.com/JanDeDobbeleer/oh-my-posh/discussions/2214

Hope my thread can help.

**Reply by ghost** [reply]
So how do you set a theme then? Because you said you don't need to set-prompt anymore.

**Reply by JanDeDobbeleer** [reply]
See [here](https://ohmyposh.dev/docs/installation/customize).
